### PR TITLE
ember-th renders only the block when one is passed.

### PR DESCRIPTION
**Pull request by cyril-sf**
It will allow to totally customize how it looks.

This will require templates to be updated if they were relying on the old behavior.
I'd also like to provide an easy way to reuse the existing html/logic to make that transition easier.

### CODE REVIEW

**Comment by mixonic** [comment]
@cyril-sf an option here might be to have the sorting logic put into a component, and have the `ember-th` yield that component for optional invocation by the block. That would provide a fairly easy adoption path for this change.

As there is no stable release I think this is ok as a breaking change. We need to fix 3.x compat before making a stable release IMO, but that is the only thing that is a blocker.

**Comment by kpfefferle** [comment]
While I like this idea in theory, I'm worried about this change breaking tables in our app that have already been converted to ET2 and expect the previous block behavior. When we feel like this has settled into a theoretical ideal, we need to compare it to the tables that have already been converted to ET2 in our own app and look for a good migration strategy.

File: addon/components/ember-th/template.hbs [hunk 1]
```diff
@@ -2,21 +2,8 @@
   {{yield columnValue columnMeta}}
 {{else}}
   {{columnValue.name}}
-{{/if}}
-
-{{#if isSorted}}
```
Comment (mixonic) [inline addon/components/ember-th/template.hbs]: The corresponding properties on the ember-th component instance should probably be dropped?

For example: https://github.com/Addepar/ember-table/blob/master/addon/components/ember-th/component.js#L78

Comment (cyril-sf) [reply addon/components/ember-th/template.hbs]: done

File: addon/components/ember-th/sort-indicator/component.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  The table header cell sort indicator component. This component renders the state of the sort on the column (ascending/descending/none).
+
+  ```hbs
+  <EmberTable as |t|>
+    <t.head @columns={{columns}} as |h|>
+      <h.row as |r|>
+        <r.cell as |columnValue columnMeta|>
+          {{columnValue.name}}
+
+          <EmberThSortIndicator columnMeta=columnMeta>
```
Comment (mixonic) [inline addon/components/ember-th/sort-indicator/component.js]: I think this should be `<EmberTh::SortIndicator @columnMeta={{columnMeta}} />`

File: tests/dummy/app/pods/docs/guides/main/table-customization/template.md [hunk 3]
```diff
@@ -42,6 +42,45 @@ lookup paths:
...
+          @widthConstraint='gte-container'
+          @fillMode='first-column'
+        
+          as |h|
+        >
+          <h.row as |r|>
+            <r.cell as |columnValue columnMeta|>
+              {{ember-th/sort-indicator columnMeta=columnMeta}}
+              Custom Header {{columnValue.name}}
+              {{ember-th/resize-handle columnMeta=columnMeta}}
```
Comment (mixonic) [inline tests/dummy/app/pods/docs/guides/main/table-customization/template.md]: These two `{{` invocations for the sort/resize components should probably also be using angle invocation.

```hbs
<EmberTh::SortIndicator @columnMeta={{columnMeta}} />
Custom Header {{columnValue.name}}
<EmberTh::ResizeHandle @columnMeta={{columnMeta}} />
```

File: tests/dummy/app/pods/docs/guides/main/table-customization/template.md [hunk 4]
```diff
@@ -42,6 +42,45 @@ lookup paths:
   {{demo.snippet name='table-customization-custom-cell-template.js' label='component.js'}}
 {{/docs-demo}}
 
+If you want to customize a header cell but still want to include the elements to sort and
+to resize a column, you can use the `ember-th/sort-indicator` and `ember-th/resize-handle`
```
Comment (mixonic) [inline tests/dummy/app/pods/docs/guides/main/table-customization/template.md]:
```suggestion
to resize a column, use the `ember-th/sort-indicator` and `ember-th/resize-handle`
```

File: addon/components/ember-th/sort-indicator/component.js (hunk 2 above)
Comment (cyril-sf) [reply addon/components/ember-th/sort-indicator/component.js]: 👍 I never used that form and couldn't find anywhere in the docs how it's supposed to be.

File: tests/dummy/app/pods/docs/guides/main/table-customization/template.md (hunk 3 above)
Comment (cyril-sf) [reply tests/dummy/app/pods/docs/guides/main/table-customization/template.md]: 👍

File: tests/integration/components/headers/ember-th-test.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        {{/ember-th}}
+      {{/ember-tr}}
+    {{/ember-thead}}
+
+    {{ember-tbody api=t rows=rows}}
+  {{/ember-table}}
+  `);
+  await firstHeader.click();
+
+  assert.equal(this.$('th [data-test-block]').length, 2, 'Header cells render passed block');
```
Comment (rondale-sc) [inline tests/integration/components/headers/ember-th-test.js]: I think mixing the page object selectors with regular selectors feels a bit weird.  Is there a way we could use the page object everywhere or not at all?

**Comment by cyril-sf** [comment]
> @cyril-sf Is this missing tests for sort indicator and resize handle?

As it's mostly a code refactor, and no other unit tests exist I didn't write any. Happy to add some.

File: tests/integration/components/headers/ember-th-test.js (hunk 5 above)
Comment (cyril-sf) [reply tests/integration/components/headers/ember-th-test.js]: Good catch 👍

**Comment by mixonic** [comment]
@cyril-sf sort indicator and resize handle are new public API, as such I expect they would be tested so we don't break them. I agree while they were implementation details they were better tested via their integration, but at least a basic set of "this public API exists" tests seem appropriate.

**Pull request merged**